src/relax/transform/fuse_tir.cc

```diff
@@ -367,17 +368,18 @@ class FusedTIRConstructor : public ExprVisitor {
    * \brief Construct a fused TIR PrimFunc from a relax sub-function
    * \param mod The IRModule
    * \param gv The global var of relax subfunction to be fused into one PrimFunc
-   * \return The fused TIR PrimFunc
+   * \return The fused TIR PrimFunc and the in-place indices (non-empty for an in-place call)
    */
-  static tir::PrimFunc GetFusedTIR(const IRModule& mod, const GlobalVar& gv) {
+  static std::pair<tir::PrimFunc, Array<Integer>> GetFusedTIR(const IRModule& mod,
+                                                              const GlobalVar& gv) {
     FusedTIRConstructor visitor(mod, gv->name_hint);
     BaseFunc f = mod->Lookup(gv);
     CHECK(f->IsInstance<relax::FunctionNode>())
         << "Expected relax functions, but got: " << f->GetTypeKey();
     CHECK(f->HasNonzeroAttr(relax::attr::kPrimitive))
         << "Expected a function with attr `kPrimitive`";
     visitor(Downcast<relax::Function>(f));
-    return visitor.fused_tir_;
+    return {visitor.fused_tir_, visitor.inplace_indices_};
```

Comment:
Can we add a safety check that all values in `visitor.inplace_indices_` are unique?  Non-unique values would mean that the fused function returns two tensors, one of which is an in-place operation that overwrites the other.

This should only be a safety check, as any failures would mean that an upstream pass had made an error by inserting `call_tir_inplace` in the first place.  However, since accidental aliasing would be rather tricky to hunt down at runtime, and since this would be an easy place to catch it, I think it would be worth adding a check.

Reply:
The entries in the buffer map are supposed to be unique anyway, but we can use a set instead to enforce that.

Reply:
Better to catch the error earlier than later, I agree.